### DISCUSSION TITLE
NaN converted to null while converting to string and back

### DISCUSSION THREAD

**Discussion by rob-pitkin** [nlohmann/json#2745] [Q&A] [answered]
I recently have been trying to handle the case where given a vector of floats, ex: [1.0, 2.0, 3.0, NAN], and I convert it into a JSON object, how to get the same vector out after converting it to a string and back. The current required steps are:
1. Create the vector of floats
2. Convert the vector into a JSON object
3. Convert that JSON object into a string (to be written to a file, transmitted, etc.)
4. Convert the resulting back into a vector of floats.

For some reason, the NaN value gets converted into a null and then never gets converted back (i.e. loss of information). Does anyone know how to fix/get around this?

**Comment by nlohmann** [comment, marked as answer]
There is no way to roundtrip NAN with this library, because JSON has no concept of a NAN.

**Reply by nlohmann** [reply]
I updated the documentation: https://json.nlohmann.me/features/types/number_handling/#nan-handling
